server/config/config.go

```diff
@@ -79,3 +90,49 @@ func (c Config) SetContextDefaultsForApp(appID apps.AppID, cc *apps.Context) *ap
 func (c Config) AppPath(appID apps.AppID) string {
 	return c.PluginURL + PathApps + "/" + string(appID)
 }
+
+func (c *Config) Reconfigure(stored StoredConfig, mmconf *model.Config) error {
+	mattermostSiteURL := mmconf.ServiceSettings.SiteURL
+	if mattermostSiteURL == nil {
+		return errors.New("plugin requires Mattermost Site URL to be set")
+	}
+	mattermostURL, err := url.Parse(*mattermostSiteURL)
+	if err != nil {
+		return err
+	}
+
+	c.StoredConfig = stored
+
+	c.MattermostSiteURL = *mattermostSiteURL
+	c.MattermostSiteHostname = mattermostURL.Hostname()
+	c.PluginURLPath = "/plugins/" + c.BuildConfig.Manifest.Id
+	c.PluginURL = strings.TrimRight(*mattermostSiteURL, "/") + c.PluginURLPath
+
+	c.MaxWebhookSize = 75 * 1024 * 1024 // 75Mb
+	if mmconf.FileSettings.MaxFileSize != nil {
+		c.MaxWebhookSize = *mmconf.FileSettings.MaxFileSize
+	}
+
+	c.DeveloperMode = pluginapi.IsConfiguredForDevelopment(mmconf)
+
+	// use CloudLambdaAccessEnvVar for now to detect the Cloud mode
+	cloudLambdaAccessKey := os.Getenv(awsapps.CloudLambdaAccessEnvVar)
+	c.CloudMode = false
+	if cloudLambdaAccessKey != "" {
+		c.CloudMode = true
+	}
+
+	if c.CloudMode {
+		c.AWSLambdaAccessKey = os.Getenv(awsapps.CloudLambdaAccessEnvVar)
```

Comment:
is `CloudMode` the MM Cloud (as opposed to Self-Managed AWS/Cloud ) mode? If yes, would it make sense to have `AWSLambdaAccessKey ` -> `APPS_LAMBDA_ACCESS_KEY` (or smth. that has no MM tie) instead of `MM_APPS_LAMBDA_ACCESS_KEY` ? (I look at the `MM` prefix here as the one that ties the var to the MM cloud, as opposed to the MM general namespace)

Reply:
Mattermost server uses `MM_` all over the place, so I assumed that this (`MM_APPS_`) is the right prefix for the new "public" settings.

For the cloud we are presently using `APPS_INVOKE_AWS_(ACCESS|SECRET)_KEY`, I left these unchanged for now. Once I find the explicit trigger for the cloud, we can use the same (MM_APPS_-prefixed) variables everywhere.

Reply:
ok, so `CloudMode` is for all clouds (MM and non-MM) and we'll use `MM_APPS_` prefix for the final/unique set of variables.

Reply:
hmmm, no :)

"CloudMode" is exclusively the Mattermost Cloud (should I rename it?) - the reverse is all "on-prem" which includes HTTP and AWS support ATM.

Reply:
Ok, it is `MattermostCloudMode` now; all else is "on-prem".

Reply:
Thanks, I got confused about the "Cloud" part, which for me included on-prem "AWS" (which to me is cloud as well - could contain Azure, other hosting types in the future), and would need similar `*ACCESS|SECRET*` keys (we could reuse the same var names, `MM_APPS` prefixed for both MM and non-MM clouds). 0/5 - but with current naming it might become harder later to distinguish between AWS and non-AWS clouds (MM or non-MM).

Reply:
I agree, the `AWS` is missing from the names, wanted to keep them short but "type of cloud" is indeed important. Changing now.

Reply:
Ah, it was `MM_APPS_S3_BUCKET` that was already there that I was trying to be consistent with. @hanzei might know what's involved in renaming it if we want a universal `MM_APPS_AWS_` prefix.

Reply:
ok, since we use one set of credentials for S3 and Lambda, I renamed them to `MM_APPS_AWS_(ACCESS|SECRET)_KEY`. I'm keeping `MM_APPS_S3_BUCKET` as is for now since it may be a dependency, but `S3` is already AWS-specific.